### ISSUE TITLE
prune for different backup archives within the same repository

### ISSUE THREAD

**Issue by languitar** [borgbackup/borg#1824]
I have struggled a long time to find out why `borg prune` with different temporal settings didn't work the way I expected it from the documentation. Basically, I am backing up several machines in a way that different parts of the same host end up in different archives, e.g. a repository might contain the following archives:
```
jaco-home-2016-11-10T09:32:11Z
jaco-home-2016-11-07T09:26:55Z
jaco-root-2016-11-10T09:41:15Z
jaco-root-2016-11-07T10:11:43Z
```
From the documentation it wasn't clear to me that `borg prune --prefix jaco- --keep...`  will treat all archives as a single collection and doesn't distinguish between different prefixes (and therefore contained files sets), which then results in daily archives arbitrarily deleting one of the two distinct prefix archives.

It would be nice if there was some special handling for such a case or at least a hint in the documentation.

**Comment by ThomasWaldmann** [comment]
There is no AI trying to make sense from archive names beyond what you tell via the `--prefix` option and this already is pointed out in the docs.

So, if you think this could be written in a better / clearer way, please give a specific suggestion for that - if possible in form of a pull request against 1.0-maint branch.

**Comment by languitar** [comment]
I've created a pull request with a proposal.

**Comment by ThomasWaldmann** [comment]
Fixed by #1830.
